### ISSUE TITLE
Code cleanup and improvments - Help needed

### ISSUE THREAD

**Issue by stevenjack** [stevenjack/cfndsl#171]
We're working to improve the code base allowing easier contributions and better test coverage.

One of the first passes on this is to get rubocop passing with a minimal set of cops, then gradually increase them as we clean the code base.

The first step has been completed and I'd like to ask anyone using `cfndsl` to try out this pre release gem (work in [this](https://github.com/stevenjack/cfndsl/pull/161) PR):

`$: gem install cfndsl --pre`

to make sure that we're not breaking any of the more complex templates people have out there. Obviously this would be covered by a test suite, although as we moved things around the code base a little we started to notice quite a few holes in the current test suite.

Please let me know of any issues you encounter in this issue.

**Comment by kornypoet** [comment]
@stevenjack For dev work, the release was cut from the `johnf_rubocop` branch right?

**Comment by stevenjack** [comment]
@kornypoet yep the pre-release gem was built from the current tip of `johnf_rubocop`.

**Comment by kornypoet** [comment]
`#addTag` was renamed `#add_tag`
I'm in favor of the new way, just noting that it changed/broke.

**Comment by stevenjack** [comment]
@kornypoet assume you guys were using it then? Funny it's really the only method that delegates off to a resource like that. We should add it back in then deprecate the method so people are made aware it'll be changing in the next major release (Which I think we'll do once we've cleaned things up and decided on anything that needs to change that would break the current API).

**Comment by kornypoet** [comment]
Yes, I think it's a good helper method. Deprecation would be safest.

**Comment by kornypoet** [comment]
Outside of that one method, I've seen no change in behavior.

**Comment by stevenjack** [comment]
@kornypoet I've pushed a new version out:

`1.5.1-pre`

**Comment by gergnz** [comment]
I can report that besides addTag -> add_tag, nothing else seems to be an issue across all our templates.

**Comment by kornypoet** [comment]
What branch was `1.5.1-pre` released from?

**Comment by stevenjack** [comment]
@kornypoet 'johnf_rubocop', the one we've been making the changes in that warranted this pre release - are you seeing unexpected behaviour?

**Comment by kornypoet** [comment]
Nothing unexpected. I have some more types to add to `aws_types.yaml` and I wanted to make the PR against the latest `pre` branch.

**Comment by kornypoet** [comment]
I think this is ready to be released as the real gem. The rubocop changes were all positive, and non-breaking (except for the already addressed `#addTag` method). There are quite a few issues and PRs waiting on this for approval.
